client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/fields.js

```diff
@@ -211,20 +223,32 @@ AddressFields.propTypes = {
 	allowChangeCountry: PropTypes.bool.isRequired,
 	errors: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ).isRequired,
 	group: PropTypes.string.isRequired,
-	countriesData: PropTypes.object.isRequired,
+	countryNames: PropTypes.object.isRequired,
+	stateNames: PropTypes.object,
 };
 
 const mapStateToProps = ( state, { group, orderId, siteId } ) => {
 	const loaded = isLoaded( state, orderId, siteId );
 	const shippingLabel = getShippingLabel( state, orderId, siteId );
-	let countriesData = getCountriesData( state, orderId, siteId );
-	if ( 'origin' === group ) {
-		countriesData = pick( countriesData, ACCEPTED_USPS_ORIGIN_COUNTRY_CODES );
+	const formData = shippingLabel.form[ group ];
+
+	let countryNames =
+		'origin' === group
+			? getOriginCountryNames( state, siteId )
+			: getDestinationCountryNames( state, siteId );
+	if ( ! countryNames[ formData.values.country ] ) {
+		// If the selected country is not supported but the user managed to select it, add it to the list
```

Comment:
Can you add some more information as to how this scenario might occur?

Reply:
I can't think of any scenario, at least in `wp-calypso`. This logic is just to be defensive against some obscure use case we accidentaly support, like changing the order address to an unsupported country in WP-Admin after having printed a label already.